ddtrace/tracer.py

```diff
@@ -300,6 +305,8 @@ def __init__(
 
         self._hooks = _hooks.Hooks()
         atexit.register(self._atexit)
+        signal.signal(signal.SIGTERM, signal_handler)
```

Comment:
This signal handler can overwrite (or be overwritten) by user's application. Here's how we wrap existing signal handlers for the serverless product: https://github.com/DataDog/dd-trace-py/blob/5183394cf2cc0eedc57398ed60e1217928db4e8a/ddtrace/contrib/aws_lambda/patch.py#L68. 

Also is registering an atexit hook sufficient? We do not intercept the a sigterm signal to send any apm events. Doing so for DSM seems a bit overkill. I worry that this could lead to some undefined behavior since we can not guarantee the state of the ddtrace library when an application is killed suddenly.

Reply:
In Kubernetes, applications are closed using a signal.
If we don't send the data to the backend in this case, we drop some payloads. Which for this use case, causes us to incorrectly track lag.
I noticed it while testing the tracer, and trying to understand why lag was tracked incorrectly after shutdown.

Reply:
I can move the signal handling to DSM only, and not the whole tracer.